packages/react/dropdown-menu/src/DropdownMenu.tsx

```diff
@@ -23,27 +25,43 @@ type DropdownMenuContextValue = {
   onOpenToggle(): void;
 };
 
+const SubmenuContext = React.createContext<boolean | undefined>(undefined);
+const InsideContentContext = React.createContext<boolean>(false);
 const [DropdownMenuProvider, useDropdownMenuContext] = createContext<DropdownMenuContextValue>(
   DROPDOWN_MENU_NAME
 );
 
 type DropdownMenuOwnProps = {
   open?: boolean;
-  defaultOpen?: boolean;
   onOpenChange?(open: boolean): void;
+  defaultOpen?: boolean;
+  dir?: Direction;
 };
 
 const DropdownMenu: React.FC<DropdownMenuOwnProps> = (props) => {
-  const { children, open: openProp, defaultOpen, onOpenChange } = props;
+  const parentSubmenuContext = React.useContext(SubmenuContext);
+  const isInsideContent = React.useContext(InsideContentContext);
+  const isRootMenu = parentSubmenuContext === undefined;
+
+  return (
+    <SubmenuContext.Provider value={!isRootMenu && isInsideContent}>
```

Comment:
I added the `isInsideContent` requirement to `DropdownMenu` as well for completeness. While I don't expect anyone needing to nest these within triggers, it at least doesn't break the code if accidentally doing so.

Reply:
Yeah I agree on this one, I don't think we should add code that isn't necessary.
I can imagine we return to this in a few months and now we wonder what it's all about, whether it actually is important or not, etc and then are afraid to remove it.